File: tests/fixtures/corpus/semgrep-positive/raw_sql_delete.py
# Corpus fixture — TRUE POSITIVE (raw-SQL deletion).
# Exercises the raw-SQL branch of missing-dsr-python: a `DELETE FROM <user table>`
# string passed to a DB cursor's execute(), in both two-arg and one-arg forms.


def purge_parameterised(cursor, uid):
    cursor.execute("DELETE FROM users WHERE id = %s", (uid,))


def purge_single_arg(cursor):
    cursor.execute("DELETE FROM user_accounts")

File: tests/fixtures/corpus/semgrep-positive/raw_sql_delete.ts
// Corpus fixture — TRUE POSITIVE (raw-SQL deletion).
// Exercises the raw-SQL branch of missing-audit-javascript and missing-dsr-javascript:
// a `DELETE FROM <user table>` string passed to a DB-execution method.

export async function purgeUser(db: any, id: string) {
  await db.query('DELETE FROM users WHERE id = $1', [id]);
}

export async function purgeViaHelper(id: string) {
  await queryWithTenantContext('DELETE FROM user_accounts WHERE id = $1', [id]);
}
